src/library-authoring/common/context.tsx

```diff
@@ -46,98 +62,136 @@ export interface LibraryContextData {
  */
 const LibraryContext = React.createContext<LibraryContextData | undefined>(undefined);
 
+interface LibraryProviderProps {
+  children?: React.ReactNode;
+  libraryId: string;
+  collectionId?: string;
+  componentPickerMode?: boolean;
+  parentLocator?: string;
+  sidebarCollectionId?: string;
+  sidebarComponentUsageKey?: string;
```

Comment:
`sidebarCollectionId` and `sidebarComponentUsageKey` are only used for testing right? I'm generally not a fan of including test-only helper code in the "main" implementation. I would prefer you made a little `<SidebarTestWrapper>` component defined in the test suite that takes `collectionId` as a prop and has code like this:

```ts
const { openCollectionInfoSidebar } = useLibraryContext();
React.useEffect(() => { openCollectionInfoSidebar(props.collectionId) });
```

Then instead of
```tsx
<LibraryProvider libraryId={library.id} sidebarCollectionId={collectionId}>{ children }</LibraryProvider>
```
you just write:
```tsx
<LibraryProvider libraryId={library.id}><SidebarTestWrapper collectionId={collectionId}>{ children }</SidebarTestWrapper></LibraryProvider>
```

But it's not a huge deal. If you stick with this approach, please just make it super clear as follows:

```suggestion
  /** Only used for testing */
  initialSidebarCollectionId?: string;
  /** Only used for testing */
  initialSidebarComponentUsageKey?: string;
```

Reply:
Done: ded2f077ec24500e65a709a2276980d3977d8bf1